src/Tempest/Http/src/RouteConfig.php

```diff
@@ -4,81 +4,22 @@
 
 namespace Tempest\Http;
 
-use Tempest\Http\Routing\Construction\MarkedRoute;
-use Tempest\Http\Routing\Construction\RoutingTree;
-use Tempest\Reflection\MethodReflector;
-
 final class RouteConfig
 {
-    /** @var string The mark to give the next route in the matching Regex */
-    private string $regexMark = 'a';
-
-    /** @var array<string, string> */
-    public array $matchingRegexes = [];
-
-    public RoutingTree $routingTree;
-
     public function __construct(
         /** @var array<string, array<string, \Tempest\Http\Route>> */
         public array $staticRoutes = [],
         /** @var array<string, array<string, \Tempest\Http\Route>> */
         public array $dynamicRoutes = [],
+        /** @var array<string, string> */
+        public array $matchingRegexes = [],
     ) {
-        $this->routingTree = new RoutingTree();
-    }
-
-    public function addRoute(MethodReflector $handler, Route $route): self
```

Comment:
This all is moved to `Routing/Construction/RouteConfigConstructor`. This RouteConfig doesn't have any behavior any more, it has become a plain old data carrying object. "unfortunatly" I couldn't make it immutable since this is treaded as a singleton we serialize, thus we need to mutate state. (I think, I could look into it more)